src/modules/dashboard/lists/SuppliedPositionsList/SuppliedPositionsList.tsx

```diff
@@ -41,16 +46,90 @@ export const SuppliedPositionsList = () => {
       })) || [];
 
   const head = [
-    <Trans key="Balance">Balance</Trans>,
-    <Trans key="APY">APY</Trans>,
-    <CollateralSwitchTooltip
-      text={<Trans>Collateral</Trans>}
-      key="Collateral"
-      variant="subheader2"
-    />,
+    {
+      title: <Trans>Asset</Trans>,
+      sortKey: 'symbol',
+    },
+    {
+      title: <Trans key="Balance">Balance</Trans>,
+      sortKey: 'underlyingBalance',
+    },
+
+    {
+      title: <Trans key="APY">APY</Trans>,
+      sortKey: 'supplyAPY',
+    },
+    {
+      title: (
+        <CollateralSwitchTooltip
+          text={<Trans>Collateral</Trans>}
+          key="Collateral"
+          variant="subheader2"
+        />
+      ),
+      sortKey: 'usageAsCollateralEnabledOnUser',
+    },
   ];
 
-  if (loading) return <ListLoader title={<Trans>Your supplies</Trans>} head={head} />;
+  if (sortDesc) {
+    if (sortName === 'symbol') {
+      suppliedPosition.sort((a, b) =>
+        a.reserve.symbol.toUpperCase() < b.reserve.symbol.toUpperCase() ? -1 : 1
+      );
+    } else if (sortName === 'usageAsCollateralEnabledOnUser') {
+      suppliedPosition.sort(
+        (a, b) =>
+          Number(a.usageAsCollateralEnabledOnUser) - Number(b.usageAsCollateralEnabledOnUser)
+      );
+    } else {
+      // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+      // @ts-ignore
+      suppliedPosition.sort((a, b) => a[sortName] - b[sortName]);
+    }
+  } else {
+    if (sortName === 'symbol') {
+      suppliedPosition.sort((a, b) =>
+        b.reserve.symbol.toUpperCase() < a.reserve.symbol.toUpperCase() ? -1 : 1
+      );
+    } else if (sortName === 'usageAsCollateralEnabledOnUser') {
+      suppliedPosition.sort(
+        (a, b) =>
+          Number(b.usageAsCollateralEnabledOnUser) - Number(a.usageAsCollateralEnabledOnUser)
+      );
+    } else {
+      // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+      // @ts-ignore
+      suppliedPosition.sort((a, b) => b[sortName] - a[sortName]);
+    }
+  }
```

Comment:
Maybe there _is_ an opportunity to dry this logic up and take in a few params to distinguish between the four lists we have.  Essentially it's only concerned with an array to sort and some key names to sort by.  I think it's achievable.  I wouldn't say it's blocking but it would clean things up maybe, and keep sorting logic consistent and more maintainable possibly.

Reply:
@drewcook ive added an idea in here 5cc90f83beff639f8d5090f36f3bdab4c94d15e4 need to still fix the types let me know if this makes more sense